PowerShellTools.HostService/PowershellServiceHostBehavior.cs

```diff
@@ -0,0 +1,87 @@
+﻿using PowerShellTools.Common.ServiceManagement;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+using System.ServiceModel.Dispatcher;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PowerShellTools.HostService
+{
+    /// <summary>
+    /// In WCF, unhandled excpetion crashes the servcie, leaving the channel into fault state, which is basically requiring client to re-instantiate proxy in order to continue using the service.
+    /// As a result, WCF provides the ability to configure a service to return information from unhandled exceptions.
+    /// This is implemetaion of the generic error handler for entire powershell wcf services, by exposing it as a service behavior attribute, so that it is developer friendly
+    /// </summary>
+    class PowershellServiceHostBehavior : Attribute, IErrorHandler, IServiceBehavior

```

Comment:
We should be explicit about the [AttributeUsage].  It looks like this is only applicable on Classes.  If so:  [AttributeUsage(AttributeTargets.Class)]